packages/engine-server/src/markdown/remark/noteRefsV2.ts

```diff
@@ -848,6 +872,23 @@ function findBeginBlockAnchor({ nodes }: { nodes: Node[] }): FindAnchorResult {
   };
 }
 
+/**
+ * Search for end of document;
+ */
+function findEndBlockAnchor({ nodes }: { nodes: Node[] }): FindAnchorResult {
+  // TODO: error if no first node found
+  const lastNode = _.last(nodes);
+  if (!lastNode) {
+    // TODO: should not happen
+    throw Error("no nodes found");
```

Comment:
Can you make the error message more specific? At least include the word 'end anchor node' or something